pkg/types/aws/machinepool.go

```diff
@@ -74,6 +80,10 @@ func (a *MachinePool) Set(required *MachinePool) {
 	if required.IAMRole != "" {
 		a.IAMRole = required.IAMRole
 	}
+
+	if required.AdditionalSecurityGroupIDs != nil {
+		a.AdditionalSecurityGroupIDs = append(required.AdditionalSecurityGroupIDs[:0:0], required.AdditionalSecurityGroupIDs...)
```

Comment:
Interesting. So this guarantees that `a.AdditionalSecurityGroupIDs` won't be `nil` if `required.AdditionalSecurityGroupIDs` has 0 elements which would be the case if we did
 ```go
a.AdditionalSecurityGroupIDs = append([]string(nil), required.AdditionalSecurityGroupIDs...)
```
We could use `copy` since we know `required.AdditionalSecurityGroupIDs` is not `nil` but it'd result in 2 lines instead of 1:
```go
a.AdditionalSecurityGroupIDs = make([]string, len(required.AdditionalSecurityGroupIDs))
copy(a.AdditionalSecurityGroupIDs, required.AdditionalSecurityGroupIDs)
```